### DISCUSSION TITLE
GraphRAG Visualizer: The Easiest Way to Visualize GraphRAG Artifacts!

### DISCUSSION THREAD

**Discussion by noworneverev** [microsoft/graphrag#884] [Community tools]
👉 [GraphRAG Visualizer](https://noworneverev.github.io/graphrag-visualizer/)<br/>
👉 [GraphRAG Visualizer Demo](https://www.youtube.com/watch?v=Hjx1iTZZtzw)

## Overview
GraphRAG Visualizer is an application designed to visualize Microsoft [GraphRAG](https://github.com/microsoft/graphrag) artifacts. By uploading parquet files generated from the GraphRAG indexing pipeline, users can easily view and analyze data without needing additional software or scripts.

## Features
- **Graph Visualization**: View the graph in 2D or 3D in the "Graph Visualization" tab.
- **Data Tables**: Display data from the parquet files in the "Data Tables" tab.
- **Search Functionality**: Fully supports search, allowing users to focus on specific nodes or relationships.
- **Local Processing**: All artifacts are processed locally on your machine, ensuring data security and privacy.
  

## Graph Data Model
The logic for creating relationships for text units, documents, communities, and covariates is derived from the [GraphRAG import Neo4j Cypher notebook](https://github.com/microsoft/graphrag/blob/community/graphrag_import_neo4j_cypher/examples_notebooks/community_contrib/neo4j/graphrag_import_neo4j_cypher.ipynb).

### Nodes

| Node | Type        |
|-----------|--------------|
| Document  | `RAW_DOCUMENT` |
| Text Unit | `CHUNK`        |
| Community | `COMMUNITY`    |
| Finding   | `FINDING`      |
| Covariate | `COVARIATE`    |
| Entity    | *Varies*       |

### Relationships

| Source Node | Relationship  | Target Node |
|-------------|---------------|-------------|
| Entity      | `RELATED`     | Entity      |
| Text Unit   | `PART_OF`     | Document    |
| Text Unit   | `HAS_ENTITY`  | Entity      |
| Text Unit   | `HAS_COVARIATE` | Covariate |
| Community   | `HAS_FINDING` | Finding     |
| Entity      | `IN_COMMUNITY` | Community  |

**Comment by CESARDELATORRE** [comment]
Questions:

- Is it possible to show the entity's and relationship's names in the visual graph?
- Is it possible to add metadata flags to the entities and then filter the visual graph to only show certain marked entities?

**Reply by noworneverev** [reply]
> Questions:
> 
> * Is it possible to show the entity's and relationship's names in the visual graph?

Yes, I’ve added options to toggle the appearance of labels for both nodes and links.

> * Is it possible to add metadata flags to the entities and then filter the visual graph to only show certain marked entities?

It’s a great idea, and it should be possible. I'll explore how to implement this.

**Comment by schauerstoff** [comment]
Unfortunately, the visualizer doesn't seem to work with graphs built in v2.0.0 or later because of the renaming of the `.parquet` files.

**Reply by noworneverev** [reply]
I've fixed it, thank you for reporting!

**Reply by CESARDELATORRE** [reply]
Cool! :)
